boot/debug.go

```diff
@@ -83,3 +87,44 @@ func DumpBootVars(w io.Writer, dir string, uc20 bool) error {
 	}
 	return nil
 }
+
+// DebugSetBootVars is a debug helper that takes a list of <var>=<value> entries
+// and sets them for the configured bootloader.
+func DebugSetBootVars(dir string, recoveryBootloader bool, varEqVal []string) error {
+	opts := &bootloader.Options{
+		NoSlashBoot: dir != "" && dir != "/",
+	}
+	if opts.NoSlashBoot || osutil.FileExists(dirs.SnapModeenvFile) {
+		// implied UC20 bootloader
+		opts.Role = bootloader.RoleRunMode
+	}
+	if recoveryBootloader {
+		// UC20 recovery bootloader
+		opts.Role = bootloader.RoleRecovery
```

Comment:
I'm confused by this code path, shouldn't we also set dir if it's not set?  it doesn't make sense I think to set Role to recovery and use dir == "/"